thunder/dynamo/splitter.py

```diff
@@ -167,6 +167,16 @@ def is_thunder_supported_partition(node: torch.fx.Node) -> bool:
         node_name = node.name
         if is_thunder_supported_partition(node):
             graph_module = getattr(split_gm, node.name)
+
+            is_differentiable_outputs = []
+            for n in graph_module.graph.nodes:
+                if n.op == "output":
+                    for n in n.all_input_nodes:
+                        if n.meta["example_value"].grad_fn is None:
```

Comment:
It's surprising that they store `grad_fn` in example_value and not just "requires_grad=True/False". Great that it works!